### ISSUE TITLE
project status, and converting modules to use rem

### ISSUE THREAD

**Issue by jsierles** [ReactExtensionManager/rem#7]
What's the status of rem? Are you guys using it internally?

As more and more modules get out there, building a standard for dependencies management is crucial. For the React Native Playground, we'd like to make automating builds easier as we push new versions up with more and more modules included.

REM looks like a great solution. One doubt: does the Cocoapod need to be published on Cocoapods, or is having a podspec enough?

To reduce developer friction, and avoid mistakes. maybe we can add an option to use fields from package.json to fill in the podspec?

Anyway,I'd like to start pushing authors to use this, via modified forks, if it's still agreed that this is the best approach.

**Comment by ide** [comment]
We're not actively working on rem or using it right now. It is a community project so if you would like to collaborate we would be happy to take PRs.

The way I currently think of rem is that it is a tool to help link native code downloaded from npm. We use CocoaPods because it is a mature tool that helps link native libraries and configure the Xcode project, but it is mostly an implementation detail that could be exchanged with another linking implementation.

To answer your technical questions, CocoaPods supports podspecs on the user's filesystem so they don't have to be published to the CocoaPods repo.

> To reduce developer friction, and avoid mistakes. maybe we can add an option to use fields from package.json to fill in the podspec?

Sounds good. I think it's best if authors don't have to write podspecs but still make it possible for advanced use cases.

> Anyway,I'd like to start pushing authors to use this, via modified forks, if it's still agreed that this is the best approach.

This would be a good way to start.

**Comment by jsierles** [comment]
Started working on this, but had a doubt: should React Native itself be installed this way using cocoapods? This could solve the problem we have today where changes to linking in React Native require manual changes in old apps.

To keep the master dependency resolver in npm, ideally we'd have react-native in package.json, marked as a native package, so its podspec is picked up by REM. Is this feasible?

**Comment by ide** [comment]
Eventually it might make sense to install React Native with rem but currently setting up a new React Native project is somewhat different from adding a native module so I think it makes more sense to focus on modules for now.

> To keep the master dependency resolver in npm, ideally we'd have react-native in package.json, marked as a native package, so its podspec is picked up by REM. Is this feasible?

Could you clarify and help me understand your question better?

**Comment by jsierles** [comment]
Makes sense!

Sorry for the confusion on the second bit. I noticed that React is included in the podspec. If using this in a project which already comes with linked libraries, like those created by ```react-native init```, would the cocoapods linking conflict?

**Comment by ide** [comment]
Sorry for the delayed response... we could exclude React from the Podfile if it going to make things complicated. The nice thing is it lets users choose how to configure React (for example: react-native init, manually linking it, or creating a custom Podfile).

**Comment by jsierles** [comment]
I agree - I'm just curious if the Pod version would override the existing linked libraries?

**Comment by ide** [comment]
I imagine the libraries would get linked twice and the app wouldn't compile, so the solution would be to remove the manually linked copy of the library.

**Comment by paramaggarwal** [comment]
I just realised that the `Podfile.spec` can actually be written as json also `Podfile.spec.json` - it is completely supported. This should make it easier to deal with it in JS and effectively generate it from the `package.json` and other files. Will try it out sometime!

**Comment by jsierles** [comment]
Very cool. Will take a look!
